spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/it/EmbeddableTreeNode.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.it;
+
+import java.util.Objects;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Embedded;
+import org.springframework.data.annotation.Id;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class EmbeddableTreeNode {
+	@Id
```

Comment:
Since this is an embedded object without a real Key, could we test this without it annotated ID ? Or perhaps just remove it from the example ? I wouldn't want a user to read that and thing that the ID annotation was required or that it had real meaning for an embedded entity.

Reply:
It is needed here, because it is used for the root node - which is the parent entity